src/client/mod.rs

```diff
@@ -25,39 +24,58 @@ pub mod builder {
 }
 
 
-pub struct Client {
-    transport: ImapTransport,
-    state: ClientState,
+pub fn connect(server: &str) -> io::Result<ImapConnectFuture> {
+    let addr = (server, 993).to_socket_addrs()?.next().ok_or_else(|| {
+        io::Error::new(io::ErrorKind::Other, format!("no IP addresses found for {}", server))
```

Comment:
I think `io::ErrorKind::AddrNotAvailable` is preferred here.

Reply:
Hmm. I'm not entirely convinced. The [documentation](https://doc.rust-lang.org/std/io/enum.ErrorKind.html) there is "A nonexistent interface was requested or the requested address was not local"; this is about finding an address for a host (not an interface), and it has nothing to do with whether the address is local or not.

Reply:
Ah. yes, sorry, for some reason I was thinking of this as binding. Never mind.